mathics/core/pattern.py

```diff
@@ -284,38 +272,37 @@ def yield_match(sub_vars, rest):
             raise StopGenerator_Pattern(True)
 
         try:
-            self.match(yield_match, expression, vars_dict, evaluation, fully=fully)
+            self.match(
+                expression=expression,
+                pattern_context={
+                    "yield_func": yield_match,
+                    "vars_dict": vars_dict,
+                    "evaluation": evaluation,
+                    "fully": fully,
+                },
+            )
         except StopGenerator_Pattern as exc:
             return exc.value
         return False
 
     def get_match_candidates(
-        self,
-        elements: Tuple[BaseElement],
-        expression: BaseElement,
-        attributes: int,
-        evaluation: Evaluation,
-        vars_dict: Optional[dict] = None,
-    ):
+        self, elements: Tuple[BaseElement], pattern_context: dict
+    ) -> tuple:
         """
-        Get the candidates that matches with the pattern.
+        Get the a sub-tuple of elements that are candidates
+        matching with the pattern.
+
+        Optional parameters provide information
+        about the context where the elements and the
+        patterns come from.
         """
         return tuple()
 
     def get_match_candidates_count(
-        self,
-        elements: Tuple[BaseElement],
-        expression: BaseElement,
-        attributes: int,
-        evaluation: Evaluation,
-        vars_dict: Optional[dict] = None,
-    ):
+        self, elements: Tuple[BaseElement], pattern_context: dict
+    ) -> Union[int, tuple]:
```

Comment:
The actual return: `return (1,1)` does match this return annotation. It should be `Tuple[int, int]`. Change will be in more_patterns_more PR>

Reply:
Regarding this, I wonder if we should not make the return value of this method a little bit more consistent, but again, it would imply making more changes about how the code works.

Reply:
I agree - the `Union[int, tuple]` should probably become `tuple``. And yes, this might be done in a future PR.